MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoReader.java

```diff
@@ -0,0 +1,273 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.media.MediaCodec;
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.util.Log;
+import android.view.Surface;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.VisibleForTesting;
+
+import com.adobe.internal.xmp.XMPException;
+
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static android.os.Build.VERSION_CODES.M;
+
+/**
+ * The MotionPhotoReader API allows developers to read through the video portion of Motion Photos in
+ * a frame-by-frame manner.
+ */
+
+public class MotionPhotoReader {
+
+    private static final String TAG = "MotionPhotoReader";
+    private final String filename;
+    private final Surface surface;
+    private final MotionPhotoInfo motionPhotoInfo;
+
+    private MediaExtractor lowResExtractor;
+    private MediaCodec lowResDecoder;
+    private MediaFormat videoFormat;
+    private FileInputStream fileInputStream;
+
+    // message keys
+    static final int MSG_NEXT_FRAME = 0x0001;
+    static final int MSG_SEEK_TO_FRAME = 0x0010;
+
+    /**
+     * Two handlers manage the calls to play through the video. The media worker thread posts
+     * available buffers to the buffer queue. The buffer worker receives messages to process frames
+     * and uses the available buffers posted by the media worker thread.
+     */
+    private HandlerThread mediaWorker;
+    private Handler mediaHandler;
+    private HandlerThread bufferWorker;
+    private BufferProcessor bufferProcessor;
+
+    /** Available buffer queues **/
+    private final BlockingQueue<Integer> availableInputBuffers;
+    private final BlockingQueue<Bundle> availableOutputBuffers;
+
+
+    /**
+     * Standard MotionPhotoReader constructor.
+     */
+    private MotionPhotoReader(String filename, Surface surface,
+                              BlockingQueue<Integer> availableInputBuffers, BlockingQueue<Bundle> availableOutputBuffers,
```

Comment:
break after `,` so that `BlockingQueue<Bundle> availableOutputBuffers,` is on its own line